### PR TITLE
Updated version based for assign devices to site to use by others

### PR DESCRIPTION
**Pull request by md-rafeek**
## Description
Added version based on the Assign devices to site common function requested by team.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [x] Tasks are idempotent (can be run multiple times without changing state)
- [ ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [x] Playbooks are modular and reusable
- [x] Handlers are used for actions that need to run on change

## Documentation
- [x] All options and parameters are documented clearly.
- [x] Examples are provided and tested.
- [x] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/module_utils/dnac.py [hunk 1]
```diff
@@ -801,39 +801,66 @@ def assign_device_to_site(self, device_ids, site_name, site_id):
...
+                    params={
+                        "site_id": site_id,
+                        "payload": device_ids
+                    },
+                )
+
+                self.check_execution_response_status(response, "assign_devices_to_site")
+                if self.status == "success":
+                    self.result["changed"] = True
+                    self.result['msg'] = "Successfully assigned site {0} to device {1}.".format(site_name, str(device_ids))
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:820]: `self.result['msg'] = "Successfully assigned device(s) {0} to site {1}.".format(str(device_ids), site_name)
`

File: plugins/module_utils/dnac.py [hunk 2]
```diff
@@ -801,39 +801,66 @@ def assign_device_to_site(self, device_ids, site_name, site_id):
...
+                    self.result['response'] = response.get("executionId")
+                    self.log(self.result['msg'], "INFO")
+                return self
+
+            except Exception as e:
+                self.msg = "Error on assign devices to site: {0}".format(str(e))
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:826]: self.msg = "Error while assigning device(s) to site: {0}".format(str(e))
also print what devices ?

File: plugins/module_utils/dnac.py [hunk 3]
```diff
@@ -801,39 +801,66 @@ def assign_device_to_site(self, device_ids, site_name, site_id):
...
                 self.log(self.msg, "ERROR")
                 self.status = "failed"
                 self.module.fail_json(msg=self.msg)
+        else:
+            assign_network_device_to_site = {
+                'deviceIds': device_ids,
+                'siteId': site_id,
+            }
+            self.log("Assigning devices to site before update: {0}, {1}".format(site_name,
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:835]: What do you mean by before update? Do you would like to say that the details of devices that are assigned to a site?

`self.log("Assigning device(s) to site '{0}' with the following details: {1}".format(site_name, str(assign_network_device_to_site)), "INFO")`

File: plugins/module_utils/dnac.py [hunk 4]
```diff
@@ -801,39 +801,66 @@ def assign_device_to_site(self, device_ids, site_name, site_id):
...
+                                                                                str(assign_network_device_to_site)), "INFO")
+            try:
+                response = self.dnac._exec(
+                    family="site_design",
+                    function='assign_network_devices_to_a_site',
+                    op_modifies=True,
+                    params=assign_network_device_to_site
+                )
+                self.log("Response from assigning devices to site: {0}, {1}, {2} .".format(
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:844]: Received API response from 'assign_network_device_to_site' while assigning devices to site: {0}, {1}, {2}

File: plugins/module_utils/dnac.py [hunk 5]
```diff
@@ -801,39 +801,66 @@ def assign_device_to_site(self, device_ids, site_name, site_id):
...
-            msg = "Failed to assign devices to site: {0}, {1}.".format(site_name,
-                                                                       str(assign_network_device_to_site))
-            self.log(msg + str(e), "ERROR")
-            site_assgin_details = str(e)
-            self.status = "failed"
-            self.module.fail_json(msg=msg, response=site_assgin_details)
+                self.check_tasks_response_status(response, api_name='assign_device_to_site')
+                if self.result["changed"]:
+                    return True
+                else:
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:850]: as we return from if.. else not required.. and make the code flow as natural..

File: plugins/module_utils/dnac.py [hunk 6]
```diff
@@ -801,39 +801,66 @@ def assign_device_to_site(self, device_ids, site_name, site_id):
...
+                    self.msg = "Failed to receive a valid response from site assignment API: {0}, {1}".format(
+                        site_name, str(assign_network_device_to_site))
+                    self.log(self.msg, "ERROR")
+                    self.status = "failed"
+                    self.module.fail_json(msg=self.msg)
+
+            except Exception as e:
+                msg = "Failed to assign devices to site: {0}, {1}.".format(
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:858]: msg = "Exception occurred while assigning devices to site '{0}'. Assignment details: {1}".format(
    site_name, str(assign_network_device_to_site))

File: plugins/module_utils/dnac.py (hunk 1 above)
Comment (md-rafeek) [reply plugins/module_utils/dnac.py]: updated

File: plugins/module_utils/dnac.py (hunk 2 above)
Comment (md-rafeek) [reply plugins/module_utils/dnac.py]: Updated the same.

File: plugins/module_utils/dnac.py (hunk 3 above)
Comment (md-rafeek) [reply plugins/module_utils/dnac.py]: This log just to print the payload before execute the function.

Comment (md-rafeek) [reply plugins/module_utils/dnac.py]: updated[](url)

File: plugins/module_utils/dnac.py (hunk 4 above)
Comment (md-rafeek) [reply plugins/module_utils/dnac.py]: updated.

File: plugins/module_utils/dnac.py (hunk 5 above)
Comment (md-rafeek) [reply plugins/module_utils/dnac.py]: updated

File: plugins/module_utils/dnac.py (hunk 6 above)
Comment (md-rafeek) [reply plugins/module_utils/dnac.py]: updated as per the message.

**Review by md-rafeek — COMMENTED** [review]
Addressed all comments

**Review by madhansansel — APPROVED** [review]

**Pull request merged**